block.go

```diff
@@ -251,6 +251,10 @@ func writeMetaFile(dir string, meta *BlockMeta) error {
 		merr.Add(f.Close())
 		return merr.Err()
 	}
+	// Force the kernel to persist the file on disk to avoid data loss if the host crashes.
+	if err := f.Sync(); err != nil {
```

Comment:
Hmm. Shouldn't you close the file in this case to avoid fd leak?

Reply:
yep! updated

Reply:
There is also a tsdb.fileutil.Sync() function which issues correct sync on macOS (default fsync doesn't issue disk barrier). It's probably better to use it.
This has been fixed in Go 1.12, so if Prometheus/TSDB require at least 1.12 it's not a big deal.

Reply:
yeah I noticed this and some parts in the code use fileutil.Fsync and others use the regular one.
I will check if there is any blocker to require 1.12 for tsdb/Prometheus and will probably open another PR to remove the workaround.